src/storage/config.rs

```diff
@@ -17,13 +17,18 @@ const DEFAULT_GC_RATIO_THRESHOLD: f64 = 1.1;
 const DEFAULT_MAX_KEY_SIZE: usize = 4 * 1024;
 const DEFAULT_SCHED_CONCURRENCY: usize = 1024 * 512;
 const MAX_SCHED_CONCURRENCY: usize = 2 * 1024 * 1024;
-const DEFAULT_RESERVER_SPACE_SIZE: u64 = 2;
+
 // According to "Little's law", assuming you can write 100MB per
 // second, and it takes about 100ms to process the write requests
 // on average, in that situation the writing bytes estimated 10MB,
 // here we use 100MB as default value for tolerate 1s latency.
 const DEFAULT_SCHED_PENDING_WRITE_MB: u64 = 100;
 
+const MIN_RESERVED_SPACE_GB: u64 = 5;
+// 20GB for reserved space is enough because size of one compaction is limited,
+// generally less than 2GB.
+pub const MAX_RESERVED_SPACE_GB: u64 = 20;
```

Comment:
I don't think it is enough for a 4TB RocksDB instance

Reply:
It's decided based on `max compaction size`, which of the default value is `2GB`. And `max-background-jobs` generally is less than 10 or 12, and some background threads are reserved for flush. So I choose `2GB * 10` as the max limit.

Reply:
2GB is not a hard limit.